src/instructlab/eval/mt_bench_common.py

```diff
@@ -273,13 +294,20 @@ def chat_completion_openai(
             output = response.choices[0].message.content
             break
         except openai.OpenAIError as e:
+            if _is_fatal_openai_error(e):
+                output = API_ERROR_OUTPUT
             if i == API_MAX_RETRY - 1:
                 # Print error on last try
                 print(type(e), e)
             else:
                 logger.debug(e)
             time.sleep(API_RETRY_SLEEP)
 
+    if output is None:
+        # Hard fail only when none of the attempts resulted in a proper
+        # response or a "soft" error (in which case we register the error
+        # in the output file).
+        raise exceptions.CompletionError(messages)
```

Comment:
This exception feels very specific for the types of errors we could be getting.  Seems useful for a 400 for example.  But probably not for a ConnectionError

Reply:
I believe you suggested to "include in error rate" for 400. If so, we won't need to raise anything at all. But yes, the class name should probably indicate it's only about basic connectivity / communication problems and such.

Reply:
Sorry, I picked a bad example.  You are correct a 400 shouldn't make it here.  My concern was the name and the messages content.  I don't know what anyone would do with the messages for the error types that would hit this.

Reply:
Yep. I am yanking the `messages` argument completely, this is of no use now that this is raised only for basic connectivity issues.